### PR TITLE
Menu - The onItemRendered event refers to an incorrect item with adaptivityEnabled (T1092214)

### PR DESCRIPTION
**Pull request by EugeniyKiyashko**

### CODE REVIEW

File: testing/tests/DevExpress.ui.widgets/menu.tests.js [hunk 1]
```diff
@@ -2950,4 +2950,72 @@ QUnit.module('itemRendered event', () => { // T906117
             assert.equal(expectedItemsArray.length, 0);
         });
     });
+
+    QUnit.test('itemRendered callback is called for menu & treeview items, adaptivityEnabled: true', function(assert) {
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets/menu.tests.js]: Don't you want to add a ticket ID to the tests' name?

File: testing/tests/DevExpress.ui.widgets/menu.tests.js [hunk 2]
```diff
@@ -2950,4 +2950,72 @@ QUnit.module('itemRendered event', () => { // T906117
...
+            expectedItemsArray.push(e);
+        };
+
+        let menu;
+        $('#menu').dxMenu({
+            dataSource: testDataSource,
+            adaptivityEnabled: true,
+            width: 50,
+            onInitialized(e) {
+                menu = e.component;
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets/menu.tests.js]: Why should we do it on initialized, not just on component creation?

File: testing/tests/DevExpress.ui.widgets/menu.tests.js [hunk 3]
```diff
@@ -2950,4 +2950,72 @@ QUnit.module('itemRendered event', () => { // T906117
             assert.equal(expectedItemsArray.length, 0);
         });
     });
+
+    QUnit.test('itemRendered callback is called for menu & treeview items, adaptivityEnabled: true', function(assert) {
+        const expectedItemsArray = [];
+        const callback = (e) => {
+            assert.equal(e.component, menu, 'component arg is menu');
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets/menu.tests.js]: Why should we do this checks exactly here? Can we use `sinon.stub()` to make all asserts in the end of the test, as by [AAA](https://automationpanda.com/2020/07/07/arrange-act-assert-a-pattern-for-writing-good-tests/)?

File: testing/tests/DevExpress.ui.widgets/menu.tests.js [hunk 4]
```diff
@@ -2950,4 +2950,72 @@ QUnit.module('itemRendered event', () => { // T906117
...
+            },
+            onItemRendered: callback
+        }).dxMenu('instance');
+
+        assert.equal(expectedItemsArray.length, 2);
+
+        assert.equal(expectedItemsArray[0].itemData.text, 'item1');
+        assert.strictEqual($(expectedItemsArray[0].itemElement).hasClass('dx-menu-item'), true);
+        assert.equal(expectedItemsArray[1].itemData.text, 'item1');
+        assert.strictEqual($(expectedItemsArray[1].itemElement).hasClass('dx-treeview-item'), true);
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets/menu.tests.js]: seems like this asserts and asserts in the previous test are the same, am i  right?

File: testing/tests/DevExpress.ui.widgets/menu.tests.js [hunk 5]
```diff
@@ -2950,4 +2950,72 @@ QUnit.module('itemRendered event', () => { // T906117
...
+            onInitialized(e) {
+                menu = e.component;
+            },
+            onItemRendered: callback
+        }).dxMenu('instance');
+
+        assert.equal(expectedItemsArray.length, 2);
+
+        assert.equal(expectedItemsArray[0].itemData.text, 'item1');
+        assert.strictEqual($(expectedItemsArray[0].itemElement).hasClass('dx-menu-item'), true);
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets/menu.tests.js]: Aren't these classes' names specified as constants in the begin of the file?

File: testing/tests/DevExpress.ui.widgets/menu.tests.js (hunk 1 above)
Comment (EugeniyKiyashko) [reply testing/tests/DevExpress.ui.widgets/menu.tests.js]: We can look at source commit name by history that contains the ticket ID. I think it's not require.

File: testing/tests/DevExpress.ui.widgets/menu.tests.js (hunk 5 above)
Comment (EugeniyKiyashko) [reply testing/tests/DevExpress.ui.widgets/menu.tests.js]: https://github.com/DevExpress/DevExtreme/pull/21782/commits/070a2b4d31e0bb460632c61c985ea59e71ac32d0

File: testing/tests/DevExpress.ui.widgets/menu.tests.js (hunk 4 above)
Comment (EugeniyKiyashko) [reply testing/tests/DevExpress.ui.widgets/menu.tests.js]: https://github.com/DevExpress/DevExtreme/pull/21782/commits/070a2b4d31e0bb460632c61c985ea59e71ac32d0

File: testing/tests/DevExpress.ui.widgets/menu.tests.js (hunk 3 above)
Comment (EugeniyKiyashko) [reply testing/tests/DevExpress.ui.widgets/menu.tests.js]: https://github.com/DevExpress/DevExtreme/pull/21782/commits/070a2b4d31e0bb460632c61c985ea59e71ac32d0

File: testing/tests/DevExpress.ui.widgets/menu.tests.js (hunk 2 above)
Comment (EugeniyKiyashko) [reply testing/tests/DevExpress.ui.widgets/menu.tests.js]: https://github.com/DevExpress/DevExtreme/commit/070a2b4d31e0bb460632c61c985ea59e71ac32d0

File: testing/tests/DevExpress.ui.widgets/menu.tests.js (hunk 1 above)
Comment (ksercs) [reply testing/tests/DevExpress.ui.widgets/menu.tests.js]: But you cannot copy this Id from history - it will be necessary to open the commit

Comment (EugeniyKiyashko) [reply testing/tests/DevExpress.ui.widgets/menu.tests.js]: https://github.com/DevExpress/DevExtreme/pull/21782/commits/719ed1660626c2c73bc8d71a45c218a85b623c43

File: testing/tests/DevExpress.ui.widgets/menu.tests.js [hunk 6]
```diff
@@ -2950,4 +2950,34 @@ QUnit.module('itemRendered event', () => { // T906117
...
+
+        $('#menu').dxMenu({
+            dataSource: testDataSource,
+            adaptivityEnabled: true,
+            width: 50,
+            onItemRendered: onItemRenderedHandler
+        }).dxMenu('instance');
+
+        const checkRenderedItem = (call, itemText, itemClass) => {
+            assert.strictEqual(onItemRenderedHandler.getCall(call).args[0].itemData.text, itemText);
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets/menu.tests.js]: we can move `onItemRenderedHandler.getCall(call).args[0]` to a variable to reduce copypaste

File: testing/tests/DevExpress.ui.widgets/menu.tests.js [hunk 7]
```diff
@@ -2950,4 +2950,34 @@ QUnit.module('itemRendered event', () => { // T906117
...
+        assert.strictEqual(onItemRenderedHandler.callCount, 2);
+        checkRenderedItem(0, 'item1', DX_MENU_ITEM_CLASS);
+        checkRenderedItem(1, 'item1', DX_TREEVIEW_ITEM_CLASS);
+
+        const $treeview = $('#menu').find(`.${DX_TREEVIEW_CLASS}`);
+        $treeview.find(`.${DX_TREEVIEW_ITEM_CLASS}`).eq(0).trigger('dxclick');
+        assert.strictEqual(onItemRenderedHandler.callCount, 3);
+        checkRenderedItem(2, 'item1_1', DX_TREEVIEW_ITEM_CLASS);
+
+        $treeview.find(`.${DX_TREEVIEW_ITEM_CLASS}`).eq(1).trigger('dxclick');
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets/menu.tests.js:2980]: we can move `$treeview.find(`.${DX_TREEVIEW_ITEM_CLASS}`)` to a variable to reduce copypaste

Comment (EugeniyKiyashko) [reply testing/tests/DevExpress.ui.widgets/menu.tests.js:2980]: We can get the second item after click on the first. So, we need additional 'find()' call to get actual items after click.

File: testing/tests/DevExpress.ui.widgets/menu.tests.js (hunk 6 above)
Comment (EugeniyKiyashko) [reply testing/tests/DevExpress.ui.widgets/menu.tests.js]: https://github.com/DevExpress/DevExtreme/pull/21782/commits/6c7393b5ca46b5cef91378fa4a6c4fc812168362

**Review by ksercs — APPROVED** [review]

**Review by Zedwag — APPROVED** [review]

**Pull request merged**